docs/guides/p4-role-configuration.rst

```diff
@@ -0,0 +1,99 @@
+partitioning the pipeline into non-overlapping sections giving each client
+exclusive access to specified P4 tables. Clients chosen for this example are
+Linux Networking and IPsec.
+
+1. Compile P4 program
+Compile the application program according to the instructions in Compiling P4
+programs guide to generate P4 artifacts for programming the pipeline.
+
+2. Extract table IDs
+From the P4 artifact `p4info.txt`, extract P4 table IDs. A helper script is
+provided to assist in extracting all table IDs from the provided input file.
+
+Each P4 client will need its own role configuration, so run the script twice to
+create two separate files.
+
+.. code-block:: text
+
+   python $P4CP_RECIPE/install/sbin/extract_table_ids_from_p4info.py \
+         -i p4info.txt -o /usr/share/stratum/ovs_p4rt_role_config.pb.txt
+
+   python $P4CP_RECIPE/install/sbin/extract_table_ids_from_p4info.py \
+         -i p4info.txt -o /usr/share/stratum/ipsec_role_config.pb.txt
+
+3. Edit role config files to partition the roles
+You will need to compare the table IDs from input file p4info.txt and the
+extracted list in role configuration files to make them exclusive. This will
+dictate each P4 client to partition the tables and gain exclusive write-access
+for the defined client role.
+
+Following shows table IDs with comments added showing table name for clarity.
+
+.. code-block:: text
+
+   $ cat /usr/share/stratum/ovs_p4rt_role_config.pb.txt
+
+   exclusive_p4_ids: 41667918  // table ID for linux_networking_control.ipv4_lpm_root_lut
+   exclusive_p4_ids: 43524029  // table ID fro linux_networking_control.ipv6_lpm_root_lut
+   exclusive_p4_ids: 40980035  // table ID for linux_networking_control.vxlan_decap_mod_table
+   exclusive_p4_ids: 37624107  // table ID for linux_networking_control.vxlan_decap_and_push_vlan_mod_table
+   exclusive_p4_ids: 42605135  // table ID for linux_networking_control.rif_mod_table_start
+   ...
+   receives_packet_ins: false
+   can_push_pipeline: true
+
+.. code-block:: text
+
+   $ cat /usr/share/stratum/ipsec_role_config.pb.txt
+
+   exclusive_p4_ids: 48773578  // table ID for MainControlDecrypt.lem_exception
+   exclusive_p4_ids: 38116127  // table ID for MainControlDecrypt.lem_clear
+   exclusive_p4_ids: 45068641  // table ID for MainControlDecrypt.ipsec_rx_sa_classification_table
+   exclusive_p4_ids: 41233864  // table ID for crypto_tunnel_control.ipsec_rx_post_decrypt_table
+   ...
+   receives_packet_ins: false
+   can_push_pipeline: false
+
+
+4. Connect P4 client to infrap4d
+Once the configuration is complete, the P4 clients will push a role name and
+the role configuration file to infrap4d with a `MasterArbitrationUpdate` message
+after the gRPC connection is initiated.
+
+Your Own P4 Client
+~~~~~~~~~~~~~~~~~~
+
+If using your own P4 client, you can set role name and pack `Any` protobuf
+message with the role configuration. Details of how to perform this are
+available on this page:
```

Comment:
```suggestion
available in this document:
```
This is a reference to an external document, not a page in the user guide.

I would reserve _page_ for the latter